gr-zeromq/lib/base_impl.cc

```diff
@@ -123,36 +123,50 @@ namespace gr {
     bool
     base_source_impl::has_pending()
     {
-      return d_msg.size() > d_consumed_bytes;
+      return !d_msgs.empty();
     }
 
     int
     base_source_impl::flush_pending(void *out_buf, const int out_nitems, const uint64_t out_offset)
     {
-      /* How much to copy in this call */
-      int to_copy_items = std::min(out_nitems, (int)((d_msg.size() - d_consumed_bytes) / d_vsize));
-      int to_copy_bytes = d_vsize * to_copy_items;
+      int total_copied_items = 0;
+      while ((total_copied_items < out_nitems) && !d_msgs.empty()) {
+        struct rx_message* msg = d_msgs.front().get();
 
-      /* Copy actual data */
-      memcpy(out_buf, (uint8_t*)d_msg.data() + d_consumed_bytes, to_copy_bytes);
+        /* How much to copy in this iteration */
+        int to_copy_items = std::min(out_nitems - total_copied_items,
+                                     (int)((msg->zmq_msg.size() - msg->consumed_bytes) / d_vsize));
+        int to_copy_bytes = d_vsize * to_copy_items;
 
-      /* Add tags matching this segment of samples */
-      for (unsigned int i=0; i<d_tags.size(); i++)
-      {
-        if ((d_tags[i].offset >= (uint64_t)d_consumed_items) &&
-            (d_tags[i].offset  < (uint64_t)d_consumed_items + to_copy_items))
+        /* Copy actual data */
+        memcpy(static_cast<uint8_t*>(out_buf) + (total_copied_items * d_vsize),
+               msg->zmq_msg.data<uint8_t>() + msg->consumed_bytes, to_copy_bytes);
```

Comment:
This is getting a compile error with g++ 4.8.4.  It's like it just needs an #include <stdint.h> at the top:

```
/opt/gnuradio/src/gr-zeromq/lib/base_impl.cc: In member function 'int gr::zeromq::base_source_impl::flush_pending(void*, int, uint64_t)':
/opt/gnuradio/src/gr-zeromq/lib/base_impl.cc:143:41: error: expected primary-expression before '>' token
                msg->zmq_msg.data<uint8_t>() + msg->consumed_bytes, to_copy_bytes);
                                         ^
/opt/gnuradio/src/gr-zeromq/lib/base_impl.cc:143:43: error: expected primary-expression before ')' token
                msg->zmq_msg.data<uint8_t>() + msg->consumed_bytes, to_copy_bytes);
```